contract/src/wallet/WalletV5R1Contract.kt

```diff
@@ -0,0 +1,253 @@
+        val data =
+            ((liteClient.getAccountState(address).account.value as? Account)?.storage?.state as? AccountActive)?.value?.data?.value?.value?.beginParse()
+        require(data != null) { throw AccountNotInitializedException(address) }
+
+        val walletData = Data.loadTlb(data)
+        walletData.walletId = walletId
+        return walletData
+    }
+
+    public suspend fun getWalletDataOrNull(walletId: WalletId): Data? = try {
+        getWalletData(walletId)
+    } catch (e: AccountNotInitializedException) {
+        null
+    }
+
+    public suspend fun transfer(
+        privateKey: PrivateKeyEd25519,
+        walletData: Data?,
+        transfer: WalletTransfer
+    ) {
+        val seqno = walletData?.seqno ?: 0
+
+        val walletData = walletData ?: Data(
+            seqno,
+            privateKey.publicKey(),
+            walletId
+        )
+
+        val stateInit = if (walletData.seqno == 0) stateInit(walletData).load() else null
+
+        val message = transferMessage(
+            address = address,
+            stateInit = stateInit,
+            privateKey = privateKey,
+            validUntil = Int.MAX_VALUE,
+            seqno = seqno,
+            walletId = walletData.walletId,
+            transfer
+        )
+
+        liteClient.sendMessage(message)
+    }
+
+    public override suspend fun transfer(
+        privateKey: PrivateKeyEd25519,
+        transfer: WalletTransfer
+    ): Unit = transfer(privateKey, getWalletDataOrNull(walletId), transfer)
+
+    public data class Data(
+        val seqno: Int,
+        val publicKey: PublicKeyEd25519,
+        var walletId: WalletId,
+        val plugins: HashMapE<Cell>
+    ) {
+        public constructor(seqno: Int, publicKey: PublicKeyEd25519, walletId: WalletId) : this(
+            seqno,
+            publicKey,
+            walletId,
+            HashMapE.empty()
+        )
+
+        public companion object : TlbConstructor<Data>(
+            "wallet.v5r1.data seqno:uint32 public_key:bits256 plugins:(HashmapE 256 (Maybe ^Cell)) = WalletV5R1Data"
+        ) {
+            override fun loadTlb(cellSlice: CellSlice): Data {
+                val authAllow = cellSlice.loadUInt(1)
+                val seqno = cellSlice.loadUInt(32).toInt()
+                val storeWalletId = cellSlice.loadInt(32)
+                val publicKey = PublicKeyEd25519(ByteString(*cellSlice.loadBits(256).toByteArray()))
+
+                // TODO wallet ID parsing
+                return Data(seqno, publicKey, WalletId(0,0,-237,0))
+            }
+
+            override fun storeTlb(cellBuilder: CellBuilder, value: Data) {
+                cellBuilder.storeUInt(1, 1) // is signature auth allowed
+                cellBuilder.storeUInt(value.seqno, 32)
+                cellBuilder.storeInt(value.walletId.serialize(), 32)
+                cellBuilder.storeBytes(value.publicKey.key.toByteArray())
+                cellBuilder.storeBoolean(false)
```

Comment:
why we dont parse it in loadTlb?

Reply:
It was because of concerns about workchain, walletVersion, and subwalletNumber of WalletID. 
We cannot use the xor operation in serialize and load the corresponding values. For now, I applied them as default values. But I'm not sure if this is a good way to do it.
https://github.com/ton-blockchain/ton-kotlin/pull/131/commits/56506f2de6194d0070c717e8a82b742de83d0630